dev/lint-repository.sh

```diff
@@ -35,4 +35,8 @@ dev/tools/check-overlays.sh || CODE=1
 echo Checking CACHEKEY
 dev/tools/check-cachekey.sh || CODE=1
 
+# Check that doc/tools/docgram/fullGrammar is up-to-date
+echo Checking grammar files
+(./configure -profile devel && make -j "$NJOBS" -f Makefile.make doc_gram_verify) || CODE=1
```

Comment:
I'm not very happy with configuring here, is there a way we could use pure dune instead?

Reply:
Not as of today, this `_verify` target is more complete than the `@check-gram` one from Dune because the latter only looks at `fullGrammar` and `orderedGrammar`, not the `rst` files.

My experience with using Dune for the documentation has been very frustrating. IMHO in the future where we have a single build system, we should remove all the Dune files for documentation and just have a small Makefile for this package (which wouldn't require calling `configure` first because it would just call Dune to build the full Coq package before).